src/Logger/LogsHttpLogger.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\logs_http\Logger\LogsHttpLogger.
+ */
+
+namespace Drupal\logs_http\Logger;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Logger\LogMessageParserInterface;
+use Drupal\Core\Logger\RfcLogLevel;
+use Drupal\Core\Logger\RfcLoggerTrait;
+use Psr\Log\LoggerInterface;
+
+class LogsHttpLogger implements LoggerInterface {
+  use RfcLoggerTrait;
+
+  /**
+   * A configuration object containing logs_http settings.
+   *
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $config;
+
+  /**
+   * The message's placeholders parser.
+   *
+   * @var \Drupal\Core\Logger\LogMessageParserInterface
+   */
+  protected $parser;
+
+  /**
+   * The severity levels array.
+   */
+  protected $severity_levels;
+
+  /**
+   * Constructs a LogsHttpLogger object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The configuration factory object.
+   * @param \Drupal\Core\Logger\LogMessageParserInterface $parser
+   *   The parser to use when extracting message variables.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, LogMessageParserInterface $parser) {
+    $this->config = $config_factory->get('logs_http.settings');
+    $this->parser = $parser;
+    $this->severity_levels = RfcLogLevel::getLevels();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function log($level, $message, array $context = array()) {
+    if ($level > $this->config->get('severity_level')) {
+      // Severity level is above the ones we want to log.
+      return;
+    }
+
+    $this->registerEvent($level, $message, $context);
+  }
+
+  /**
+   * Register an event in a static cache.
+   *
+   * To prevent multiple registration of the same error, we check that identical
+   * events are not captured twice, thus reducing the final HTTP requests needed.
+   *
+   * @param $level
+   *  The severity level.
+   * @param message
+   *  The message that contains the placeholders.
+   * @param array $context
+   *  The context as passed from the main Logger.
+   */
+  protected function registerEvent($level, $message, array $context = array()) {
+    if (!logs_http_get_http_url()) {
```

Comment:
Basically yes, but we also use it on `logs_http_shutdown()` and i'm not sure we can convert the `logs_http_shutdown` to be a method as well...